include/envoy/network/address.h

```diff
@@ -16,6 +16,22 @@ namespace Envoy {
 namespace Network {
 namespace Address {
 
+/**
+ * Result holds the rc and errno values resulting from a system call.
+ */
+struct Result {
```

Comment:
Q: Is this really address specific? Should we potentially put this in https://github.com/envoyproxy/envoy/blob/master/include/envoy/api/os_sys_calls.h as SyscalResult or something like that? I don't feel that strongly but throwing it out there.

Reply:
That makes perfect sense. I wanted to use this `struct` in other places (`buffer`, etc.) anyway.

Reply:
Addressed in bb2841e.